### PR TITLE
added published timestamp

### PR DESCRIPTION
**Pull request by TGS963**
## Changes

- Added a global state for whether to publish (change publishedAt time) or not.
- Added display of publishedAt on notes public page
- Removed redundant database calls

### CODE REVIEW

File: components/dashboard/TextArea/PostButtons.tsx [hunk 1]
```diff
@@ -71,16 +71,19 @@ const PostButtons = ({ shiftRight, editorRef }: PostButtonsProps) => {
...
-      public: postPublic,
-    });
+    await updateNote(
+      {
+        id: selectedNoteId,
+        title: postTitle,
+        content: postContent,
+        public: postPublic,
+      },
+      publish
```
Comment (NayamAmarshe) [inline components/dashboard/TextArea/PostButtons.tsx:85]: We don't need this here. We can pass it in the updateNote function.

File: components/hooks/useNotes.ts [hunk 2]
```diff
@@ -62,33 +62,30 @@ export const useNotes = ({ userId }: UseNotesProps) => {
...
-      public?: boolean;
-    }) => {
+    async (
+      note: {
+        id: string;
+        title: string;
+        content: string;
+        public?: boolean;
+      },
+      publish?: boolean
```
Comment (NayamAmarshe) [inline components/hooks/useNotes.ts:72]:
```suggestion
      }
```

File: components/hooks/useNotes.ts [hunk 3]
```diff
@@ -62,33 +62,30 @@ export const useNotes = ({ userId }: UseNotesProps) => {
...
       if (!userId || !note) return;
 
       const notesRef = doc(db, "users", userId, "notes", note.id);
-      const publicRef = doc(db, "public_notes", note.id);
       const currentTime = new Date().getTime();
-
-      const publicNote = await getDoc(publicRef);
+      const updatedContent = publish
+        ? { ...note, updatedAt: currentTime, publishedAt: currentTime }
+        : { ...note, updatedAt: currentTime };
```
Comment (NayamAmarshe) [inline components/hooks/useNotes.ts:80]:
```suggestion
      const updatedContent = !public
        ? { ...note, updatedAt: currentTime, publishedAt: currentTime }
        : { ...note, updatedAt: currentTime };
```

File: pages/[username]/posts/[postId].tsx [hunk 4]
```diff
@@ -78,6 +78,7 @@ export const PostPage = ({ note, name, profilePicture }: Props) => {
               <h1 className="text-center text-5xl font-bold leading-tight">
                 {note.title}
               </h1>
+              <p>Last Published: {formatNumber(note.publishedAt)}</p>
```
Comment (NayamAmarshe) [inline pages/[username]/posts/[postId].tsx:81]: We won't be needing this here. We can show the date inside the publish modal but no need for it on the main screen.

**Review by NayamAmarshe — CHANGES_REQUESTED** [review]

File: pages/[username]/posts/[postId].tsx (hunk 4 above)
Comment (TGS963) [reply pages/[username]/posts/[postId].tsx:81]: Got it!

File: components/hooks/useNotes.ts (hunk 3 above)
Comment (TGS963) [reply components/hooks/useNotes.ts:80]: Won't it be `note.public`?

Also, I should've done it in the first place. It was unnecessary to modify the updateNotes call

**Review by NayamAmarshe — APPROVED** [review]

**Pull request merged**